### PR TITLE
feat(commitment): per-bucket commitment + pricing on TimeOfDayBucket

### PR DESCRIPTION
**Pull request by DiabolusGX**
## Summary

Extends `TimeOfDayBucket` so each bucket carries its own commitment (amount/quantity, overage factor, true-up) and its own SUBSCRIPTION-scoped base price. Out-of-bucket time falls back to the line item's existing commitment + price. Per-bucket math flows through billing (correct totals) and analytics (opt-in per-point bucket identity + bucket-level summaries). Invoice DTO is unchanged (aggregate-only).

Spec: `docs/superpowers/specs/2026-06-08-time-of-day-bucket-commitment-design.md`
Plan: `docs/superpowers/plans/2026-06-08-time-of-day-bucket-commitment.md`

## What's in this PR

**Type + validation layer**
- Extended `TimeOfDayBucket` with `ID`, `PriceID`, `CommitmentType`, `CommitmentValue`, `OverageFactor`, `TrueUpEnabled` + `HasCommitment()` helper.
- Per-bucket field `Validate()` and array-level `ValidateNoOverlap` (with midnight-wrap) + `ValidateWindowAlignment` validators.

**DTO + service layer**
- `CommitmentBucketRequest` DTO with inline `CreatePriceRequest` per bucket; retyped `CommitmentTimeBuckets` on create/update DTOs.
- `resolveBucketPrices` service helper materializes one SUBSCRIPTION-scoped price per bucket + UUID per bucket inside the existing transaction.
- Wired into `AddSubscriptionLineItem` (reachable from `CreateSubscription`'s line-item flow) and `UpdateSubscriptionLineItem` (replace-all semantics).
- Array validation invoked from service paths with meter-window lookup.
- Cumulative-commitment + per-bucket combination rejected at validation.

**Billing math**
- `chargeAtQty` + `bucketIndexAtWindowStart` helpers.
- `applyWindowCommitmentToLineItem` split into `applyWindowCommitmentLegacy` (verbatim existing body) and `applyWindowCommitmentPerBucket` (new); dispatcher selects based on `HasCommitment()` + `PriceID`.
- Unified true-up: `price_at_qty(commitment) − price_at_qty(actual)`, tier-aware via existing `priceService.CalculateCost`.

**Analytics**
- Opt-in `breakdown_bucket` flag on usage analytics — per-point `BucketID` / `bucket_price_id` stamping + `BucketSummaries` array with utilized / overage / true-up rollups.
- No ClickHouse query builder changes — augmentation is post-aggregation in Go.

## Scope decisions worth flagging

- **Addons:** Inline-bucket flow not in current addon DTOs. Addons attaching new bucketed line items would require new addon-DTO fields + persistence story — out of scope here. Pre-materialized bucket support via `LineItemCommitmentConfig` still works as before.
- **Scheduled-successor lock on update:** No "successor line item" concept exists in the codebase today. `UpdateSubscriptionLineItem` ships replace-all semantics (request buckets replace stored buckets; old prices archived). A scheduled-successor lock can layer on later if/when that concept is introduced.
- **Integration test:** Project has no testcontainers infrastructure today. The end-to-end test runs against the existing in-memory `BaseServiceTestSuite` and exercises the full service path + commitment math + sum invariant. A testcontainers variant could add persistence + ClickHouse aggregation coverage later.

## Test plan

- [x] `go test ./internal/types/... ./internal/api/dto/... ./internal/service/...` — all pass
- [x] `go build ./internal/... ./cmd/...` — clean
- [x] `go vet ./internal/... ./cmd/...` — clean
- [x] Legacy bucket shape (`{Start, End}` only, no per-bucket fields) bills bit-identically via the legacy path (regression test `TestApplyWindowCommitment_LegacyBucketShape`).
- [x] Per-bucket dispatch verified with flat-fee + quantity-commitment scenarios; sum invariant asserted.
- [x] End-to-end test asserts grand total $840 with $320 true-up, zero overage, sum invariant holds.
- [ ] Manual: run swagger regen (`make swagger`) before merge.
- [ ] Manual: run SDK regen (`make go-sdk`) before merge.
- [ ] Manual: smoke-test the analytics `breakdown_bucket` flag via the HTTP API.

## Files touched

15 files, +2906 / −33:

- `internal/types/commitment.go` + tests
- `internal/types/commitment_bucket_validation.go` (new) + tests
- `internal/api/dto/subscription_line_item.go` + tests, `internal/api/dto/subscription.go`, `internal/api/dto/events.go`
- `internal/service/subscription_line_item.go` + tests
- `internal/service/subscription.go` + tests
- `internal/service/billing_commitment.go` + tests
- `internal/service/feature_usage_tracking.go` + new `feature_usage_tracking_bucket_test.go`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Time-of-day commitment buckets for subscription line items with per-bucket pricing, per-point bucket stamping, and an opt-in analytics bucket breakdown producing per-bucket summary rows.

* **Improvements**
  * Automatic creation of subscription-scoped bucket prices on create/update, stronger bucket validation (overlap/alignment/meter checks), and more accurate billing where commitment handling applies more broadly.

* **Tests**
  * Extensive unit and integration tests covering bucket behavior, pricing, analytics, and edge cases.

### CODE REVIEW

File: internal/service/subscription_line_item.go [hunk 1]
```diff
@@ -833,15 +915,152 @@ func (s *subscriptionService) validateMultiCadence(sub *subscription.Subscriptio
...
+					WithReportableDetails(map[string]interface{}{"bucket_id": reqs[i].ID, "bucket_index": i}).
+					Mark(ierr.ErrValidation)
+			}
+			buckets[i].PriceID = ex.PriceID
+			continue
+		}
+
+		priceReq := *reqs[i].Price
+		priceReq.EntityType = types.PRICE_ENTITY_TYPE_SUBSCRIPTION
+		priceReq.EntityID = subscriptionID
```
Comment (DiabolusGX) [inline internal/service/subscription_line_item.go:1034]: should we create new entity type `BUCKET` to represent that this price is for given bucket.
optionally: add bucket it in price metadata.

**Review by nkmishra1997 — APPROVED** [review]
